docs/user/openstack/README.md

```diff
@@ -49,6 +49,94 @@ hang waiting for the console to come up.
+HA) to an external load balancer.
+
+The load balancer must serve ports 6443, 443, and 80 to any users of
+the system.  Port 49500 is for serving ignition startup configurations
+to the OpenShift nodes and should not be reachable by the outside world.
+
+The first step is to add floating IPs to all the master nodes. Usually
+the public network here is named 'public':
+
+* `openstack floating ip create --port master-port-0 <public network>`
+* `openstack floating ip create --port master-port-1 <public network>`
+* `openstack floating ip create --port master-port-2 <public network>`
+
+Once complete you can see your floating IPs using:
+
+* `openstack server list`
+
+These floating IPs can then be used by the load balancer to access
+the cluster.  An example haproxy configuration for port 6443 is below.
+The other port configurations are identical.
+
+```
+listen <cluster name>-api-6443
+    bind 0.0.0.0:6443
+    mode tcp
+    stats enable
+    stats uri /haproxy?status
+    balance roundrobin
+    server ostest-master-2 <floating ip>:6443 check
+    server ostest-master-0 <floating ip>:6443 check
+    server ostest-master-1 <floating ip>:6443 check
+```
+
+Next step is to allow access to the network the load balancer is on:
+
+* `openstack security group rule create master --remote-ip <subnet CIDR> --ingress --protocol tcp --dst-port 6443`
+* `openstack security group rule create master --remote-ip <subnet CIDR> --ingress --protocol tcp --dst-port 443`
+* `openstack security group rule create master --remote-ip <subnet CIDR> --ingress --protocol tcp --dst-port 80`
+
+Where subnet CIDR is the network the load balancer is on.  You could
+also specify a specific IP address with /32 if you wish.
+
+You can verify the operation of the load balancer now if you wish, using the
+curl commands given below.
+
+Now the DNS entry for <cluster name>-api.<base domain> needs to be updated
+to point to the new load balancer.  In our case the cluster name is
+'ostest' and the domain is 'shiftstack.com':
+
+* `<load balancer ip> ostest-api.shiftstack.com`
+
+The external load balancer should now be operation along with your own
+DNS solution.  It's best to test this configuration before removing
+the api. The following curl command is an example of how
+to check functionality:
+
+`curl https://<loadbalancer-ip>:6443/version --insecure`
+
+Result:
+
+```json
+{
+  "major": "1",
+  "minor": "11+",
+  "gitVersion": "v1.11.0+ad103ed",
+  "gitCommit": "ad103ed",
+  "gitTreeState": "clean",
+  "buildDate": "2019-01-09T06:44:10Z",
+  "goVersion": "go1.10.3",
+  "compiler": "gc",
+  "platform": "linux/amd64"
+}
+```
+
+Another useful thing to check is that the ignition configurations are only
+available from within the deployment:
+
+* `curl https://<loadbalancer ip>:49500/config/master --insecure`
+
+Now that the DNS and load balancer has been moved, we can take down the existing
+api VM:
+
+* `openstack server delete <cluster name>-api`
+
```

Comment:
What do you think about adding a note regarding the master's scale-out process? A master's scale-out, should there be one, will require repeating the above steps for that master instance only. Sounds kinda obvious but explicit is better than implicit, I guess :smile:

Reply:
I don't think that's fully supported right now anyway

Reply:
I wouldn't wait for the scale-out to be supported to add a line like this. It sounds obvious but there's no harm in documenting that.